pixeltable/share/packager.py

```diff
@@ -200,6 +220,92 @@ def __build_tarball(self) -> Path:
                 tf.add(src_file, arcname=f'media/{dest_name}')
         return bundle_path
 
+    def __extract_preview_data(self) -> tuple[dict[str, str], list[list[Any]]]:
+        """
+        Extract a preview of the table data for display in the UI.
+
+        In order to bound the size of the output data, all "unbounded" data types are resized:
+        - Strings are abbreviated as per Formatter.abbreviate()
+        - Arrays and JSON are shortened and formatted as strings
+        - Images are resized to thumbnail size as a base64-encoded webp
+        - Videos are replaced by their first frame and resized as above
+        - Documents are replaced by a thumbnail as a base64-encoded webp
+        """
+        # First 8 columns
```

Comment:
Why not include all columns? That gives the UI more flexibility (meaning we don't have to re-write bundles if we decide to show more columns or a different subset), and there isn't really a downside.

Reply:
This is all part of the synchronous API call, and it will be loaded when the table is browsed, so we're really trying to bound the data size.

If a table has 50 columns and we do 50 columns x 20 rows and there are a lot of images, the request could get pretty fat.